server/src/main/java/org/apache/iotdb/db/tools/TsFileSplitTool.java

```diff
@@ -0,0 +1,257 @@
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class TsFileSplitTool {
+
+  private static final Logger logger = LoggerFactory.getLogger(TsFileSplitTool.class);
+
+  private final String filename;
+
+  /**
+   * If the chunk point num is lower than this threshold, it will be deserialized into points,
+   * default is 100
+   */
+  private final long chunkPointNumLowerBoundInCompaction =
+      IoTDBDescriptor.getInstance().getConfig().getChunkPointNumLowerBoundInCompaction();
+
+  private final double targetSplitFileSize =
+      IoTDBDescriptor.getInstance().getConfig().getTargetCompactionFileSize();
+
+  /** Maximum index of plans executed within this TsFile. */
+  protected long maxPlanIndex = Long.MIN_VALUE;
+
+  /** Minimum index of plans executed within this TsFile. */
+  protected long minPlanIndex = Long.MAX_VALUE;
+
+  public static void main(String[] args) throws IOException {
+    String fileName = args[0];
+    logger.info("Splitting TsFile {} ...", fileName);
+    new TsFileSplitTool(fileName).run();
+  }
+
+  /* construct TsFileSketchTool */
+  public TsFileSplitTool(String filename) {
+    this.filename = filename;
+  }
+
+  /* entry of tool */
+  @SuppressWarnings("squid:S3776") // Suppress high Cognitive Complexity warning
+  public void run() throws IOException {
+    TsFileIOWriter writer = null;
+
+    try (TsFileSequenceReader reader = new TsFileSequenceReader(filename)) {
+      Iterator<List<Path>> pathIterator = reader.getPathsIterator();
+      Set<String> devices = new HashSet<>();
+      String[] filePathSplit = filename.split(IoTDBConstant.FILE_NAME_SEPARATOR);
+      int versionIndex = Integer.parseInt(filePathSplit[filePathSplit.length - 3]) + 1;
+      // to avoid compaction after restarting. NOTICE: This will take effect only in
+      filePathSplit[filePathSplit.length - 2] = "10";
+      // server v0.12
+
+      while (pathIterator.hasNext()) {
+        for (Path path : pathIterator.next()) {
+          String deviceId = path.getDevice();
+          if (devices.add(deviceId)) {
+            if (writer != null && writer.getPos() < targetSplitFileSize) {
+              writer.endChunkGroup();
+              writer.startChunkGroup(deviceId);
+            } else {
+              if (writer != null) {
+                // seal last TsFile
+                TsFileResource resource = endFileAndGenerateResource(writer);
+                resource.close();
+              }
+
+              filePathSplit[filePathSplit.length - 3] = String.valueOf(versionIndex);
+              StringBuilder sb = new StringBuilder();
+              for (int i = 0; i < filePathSplit.length; i++) {
+                sb.append(filePathSplit[i]);
+                if (i != filePathSplit.length - 1) {
+                  sb.append("-");
```

Comment:
"-"

Is it in Constant?

Reply:
Fixed to `IoTDBConstant.FILE_NAME_SEPARATOR`